pulp_rpm/app/models.py

```diff
@@ -1,33 +1,203 @@
 
-    # field1 = models.TextField(blank=False, null=False)
-    # field2 = models.TextField(blank=False, null=False)
+    # A string containing a JSON-encoded list of dictionaries, each of which represents a single
+    # file. Each file dict contains the following fields:
+    #
+    #   type (str):     one of "" (regular file), "dir", "ghost"
+    #   path (str):     path to file
+    #   name (str):     filename
+    files = models.TextField(default='[]', blank=True)
 
-    # class Meta:
-    #     unique_together = (
-    #         'field1',
-    #         'field2'
-    #     )
+    # Each of these is a string containing a JSON-encoded list of dictionaries, each of which
+    # represents a dependency. Each dependency dict contains the following fields:
+    #
+    #   name (str):     name
+    #   flags (str):    flags
+    #   epoch (str):    epoch
+    #   version (str):  version
+    #   release (str):  release
+    #   pre (bool):     preinstall
+    requires = models.TextField(default='[]', blank=True)
+    provides = models.TextField(default='[]', blank=True)
+    conflicts = models.TextField(default='[]', blank=True)
+    obsoletes = models.TextField(default='[]', blank=True)
+    suggests = models.TextField(default='[]', blank=True)
+    enhances = models.TextField(default='[]', blank=True)
+    recommends = models.TextField(default='[]', blank=True)
+    supplements = models.TextField(default='[]', blank=True)
+
+    location_base = models.TextField(blank=True)
+    location_href = models.TextField(blank=True)
+
+    rpm_buildhost = models.TextField(blank=True)
+    rpm_group = models.TextField(blank=True)
+    rpm_license = models.TextField(blank=True)
+    rpm_packager = models.TextField(blank=True)
+    rpm_sourcerpm = models.TextField(blank=True)
+    rpm_vendor = models.TextField(blank=True)
+    rpm_header_start = models.BigIntegerField(null=True, blank=True)
+    rpm_header_end = models.BigIntegerField(null=True, blank=True)
+
+    size_archive = models.BigIntegerField(null=True, blank=True)
+    size_installed = models.BigIntegerField(null=True, blank=True)
+    size_package = models.BigIntegerField(null=True, blank=True)
+
+    time_build = models.BigIntegerField(null=True, blank=True)
+    time_file = models.BigIntegerField(null=True, blank=True)
+
+    @property
+    def nevra(self):
+        """ Package NEVRA string (Name-Epoch-Version-Release-Architecture)
+        """
+        return "{n}-{e}:{v}-{r}.{a}".format(
+            n=self.name, e=self.epoch, v=self.version, r=self.release, a=self.arch)
+
+    @property
+    def nvra(self):
+        """ Package NVRA string (Name-Version-Release-Architecture)
+        """
+        return "{n}-{v}-{r}.{a}".format(
+            n=self.name, v=self.version, r=self.release, a=self.arch)
+
+    class Meta:
+        unique_together = (
+            'name', 'epoch', 'version', 'release', 'arch', 'checksum_type', 'pkgId'
+        )
+
+
+class SrpmContent(RpmContent):
```

Comment:
Should this be a separate content type, or an ```is_srpm=True``` boolean flag on the RPM model?

Reply:
I'm personally +1 to have a separate type. It will be a separate table in DB, right? I think it will make all the type related operations easier.

Reply:
Regardless of how it's modeled. it would be great if when filtering packages that you have 1 endpoint that has both rpms and srpms with a flag like `is_srpm=True`. With that in mind modeling it as the same may be the easiest. I'm ok with either approach.